src/gt4py/next/ffront/decorator.py

```diff
@@ -351,6 +359,61 @@ def _column_axis(self):
         return iter(scanops_per_axis.keys()).__next__()
 
 
+@dataclasses.dataclass(frozen=True)
+class ProgramWithBoundArgs(Program):
+    bound_args: dict[str, typing.Union[float, int, bool]] = None
+
+    def _process_args(self, args: tuple, kwargs: dict):
+        args = list(args)
+        b_args = list(self.bound_args.keys()) + args
+        param_ids = [param.id for param in self.past_node.params]
+
+        for name in self.bound_args.keys():
+            if name in kwargs:
+                # TODO(tehrengruber): use error with source location
+                raise RuntimeError(f"Parameter `{name}` already set as a bound argument.")
```

Comment:
```suggestion
                raise TypeError_(self.past_node.loc, f"Parameter `{name}` already set as a bound argument.")
```
Please raise a `gt4py.next.errors.TypeError_` and below.